test-case/check-alsabat.sh

```diff
@@ -0,0 +1,87 @@
+#!/bin/bash
+
+##
+## Case Name: check alsabat
+## Preconditions:
+##    This test case requires physical loopback between playback and capture.
+##    playback <=====>  capture
+##    nocodec : no need to use hw loopback cable, It support DSP loopback by quirk
+## Description:
+##    run alsabat test on the specified pipelines
+## Case step:
+##    1. Specify the pcm IDs for playback and catpure
+##    3. run alsabat test
+## Expect result:
+##    The return value of alsabat is 0
+##
+
+# remove the existing alsabat wav files
+rm -f /tmp/bat.wav.*
+
+source $(dirname ${BASH_SOURCE[0]})/../case-lib/lib.sh
+
+OPT_OPT_lst['p']='pcm_p'     	OPT_DESC_lst['p']='pcm for playback. Example: hw:0,0'
+OPT_PARM_lst['p']=1          	OPT_VALUE_lst['p']=''
+
+OPT_OPT_lst['c']='pcm_c'      	OPT_DESC_lst['c']='pcm for capture. Example: hw:1,0'
+OPT_PARM_lst['c']=1             OPT_VALUE_lst['c']=''
+
+OPT_OPT_lst['f']='frequency'    OPT_DESC_lst['f']='target frequency'
+OPT_PARM_lst['f']=1             OPT_VALUE_lst['f']=997
+
+OPT_OPT_lst['n']='frames'     OPT_DESC_lst['n']='test frames'
+OPT_PARM_lst['n']=1             OPT_VALUE_lst['n']=240000
+
+OPT_OPT_lst['s']='sof-logger'   OPT_DESC_lst['s']="Open sof-logger trace the data will store at $LOG_ROOT"
+OPT_PARM_lst['s']=0             OPT_VALUE_lst['s']=1
+
+func_opt_parse_option $*
+
+pcm_p=${OPT_VALUE_lst['p']}
+pcm_c=${OPT_VALUE_lst['c']}
+frequency=${OPT_VALUE_lst['f']}
+frames=${OPT_VALUE_lst['n']}
+
+if [ "$pcm_p" = "" ]||[ "$pcm_c" = "" ];
+then
+	dloge "No playback or capture PCM is specified. Skip the alsabat test"
+	exit 2
+fi
+
+[[ ${OPT_VALUE_lst['s']} -eq 1 ]] && func_lib_start_log_collect
+
+platform=$(sof-dump-status.py -p)
+
+function __upload_wav_file
+{
+    # upload the alsabat wav file
+    for file in /tmp/bat.wav.*
+    do
+	    size=`ls -l $file | awk '{print $5}'`
+	    if [[ $size -gt 0 ]]; then
+		    cp $file $LOG_ROOT/
+	    fi
+    done
+}
+
+dlogi "platform is $platform"
```

Comment:
is the platform needed to output here?

Reply:
Deleted. This output is not necessary in this case.